### PR TITLE
Added ability to accept certificates without fingerprint verification

### PR DESCRIPTION
**Pull request by benbooth493**
This has been implemented by adding the --accept-certificate flag to
'lxc remote add'
Signed-off-by: Ben Booth <benbooth493@gmail.com>

### CODE REVIEW

File: lxc/remote.go [hunk 1]
```diff
@@ -24,18 +26,20 @@ func (c *remoteCmd) usage() string {
...
 		"Manage remote LXD servers.\n" +
 			"\n" +
-			"lxc remote add <name> <url>        Add the remote <name> at <url>.\n" +
-			"lxc remote remove <name>           Remove the remote <name>.\n" +
-			"lxc remote list                    List all remotes.\n" +
-			"lxc remote rename <old> <new>      Rename remote <old> to <new>.\n" +
-			"lxc remote set-url <name> <url>    Update <name>'s url to <url>.\n" +
-			"lxc remote set-default <name>      Set the default remote.\n" +
-			"lxc remote get-default             Print the default remote.\n")
+			"lxc remote add <name> <url> --accept-certificate  Add the remote <name> at <url>.\n" +
```
Comment (chrisglass) [inline lxc/remote.go]: Shouldn't that be between "[]" since it's optional?

Comment (benbooth493) [inline lxc/remote.go]: It should...my bad.

**Comment by chrisglass** [comment]
You need the Signed-off-by bits to all of your commits... "git commit -s" can generate them for you, FWIW.

**Comment by benbooth493** [comment]
Ok np. Will bear that in mind.

**Comment by chrisglass** [comment]
Another option (although sometimes frowned upon) is to squash your commits locally (so only Signed-off-by commits are visible) and push your branch with --force.

It rewrites public history, which is sometimes evil, but it makes it a bit easier to work in this particular workflow (personal opinion)

File: lxc/remote.go [hunk 2]
```diff
@@ -111,6 +115,10 @@ func addServer(config *lxd.Config, server string, addr string) error {
 		return nil
 	}
 
+	if acceptCert {
+		c.AcceptCert = true
```
Comment (tych0) [inline lxc/remote.go]: Can we make this an argument to UserAuthServerCert vs. a member on the Client struct? I think it's only relevant for a single call to the function, vs. all subsequent calls.

**Comment by benbooth493** [comment]
I have been squashing locally to be honest, but I'll run git commit -s from now on.
Thanks

File: lxc/remote.go (hunk 2 above)
Comment (benbooth493) [inline lxc/remote.go]: Sure np

**Comment by benbooth493** [comment]
I have no idea why I'm having so much trouble with po/lxd.pot
I'm running 'make i18n' before every commit.

Can anyone shed any light on this?

**Comment by chrisglass** [comment]
Every time you touch a string, the po files need to be regenerated. Travis
is aware of that and will therefore fail if you forget to regenerate the po
files.
Also, po files have a "date generated" field... it makes things a bit
awkward.

Arguably, the bot should actually be doing it for us instead of bossing the
humans around... but that's another topic :)

**Comment by benbooth493** [comment]
I've been rebuild the po files every time but I still get test errors.
What's your workflow regarding this?

**Comment by chrisglass** [comment]
Oh! It's failing because https://github.com/benbooth493/lxd/commit/e3dfd97caffd15381f0133ef29f53cd9cdde880f has no "Signed-off-by" :(

Same deal with https://github.com/benbooth493/lxd/commit/444ccea4a3369ec1cd822da924b08baa004125bd

I wish we could overlook this rule, but it's unfortunately very important that commits are acknowledged this way (this stackoverflow answer might sched some more light on the issue it solves: https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

**Comment by benbooth493** [comment]
Of course! Thanks!

**Comment by benbooth493** [comment]
ftw!

**Comment by chrisglass** [comment]
\o/

(I can't merge though, I don't have commit bit here :) )

**Comment by hallyn** [comment]
Thanks!

**Comment by benbooth493** [comment]
This feature's intention was to accept the certificate fingerprint unconditionally rather than skip certificate verification.

Maybe my feature is really "accept-fingerprint"?

**Pull request merged**